packages/element/src/create-interpolate-element.js

```diff
@@ -0,0 +1,355 @@
+/**
+ * External dependencies
+ */
+import { createElement, cloneElement, Fragment, isValidElement } from 'react';
+
+let indoc,
+	offset,
+	output,
+	stack;
+
+/**
+ * Matches tags in the localized string
+ *
+ * This is used for extracting the tag pattern groups for parsing the localized
+ * string and along with the map converting it to a react element.
+ *
+ * There are four references extracted using this tokenizer:
+ *
+ * match: Full match of the tag (i.e. <strong>, </strong>, <br/>)
+ * isClosing: The closing slash, it it exists.
+ * name: The name portion of the tag (strong, br) (if )
+ * isSelfClosed: The slash on a self closing tag, if it exists.
+ *
+ * @type {RegExp}
+ */
+const tokenizer = /<(\/)?(\w+)\s*(\/)?>/g;
+
+/**
+ * Describes an element and it's properties.
+ *
+ * This is used by the string iterator to track children that get added to an
+ * element when it is created. This allows for collecting nested elements in
+ * the string before creating the parent.
+ *
+ * @private
+ *
+ * @param {string}    name    The name of the element.
+ * @param {WPElement} element The element
+ *
+ * @return {Element} A constructor for the Element.
+ */
+function Element( name, element ) {
```

Comment:
ya it definitely introduces some mental soup when you put it that way.  I'll see if I can come up with something a little nicer.

Reply:
So bear in mind that this is a private function (so there's some play here in later refactors) and in acknowledging the mental soup of having `Element.element` and `Element.children`, I decided to come up with a name that sort of describes what this thing is and is distinct enough for following the flow.  I chose `Filament` because a filament in the natural world (in certain contexts) can be considered a conductor of something.  Since this thing is a conductor of element and it's children (functionally) I thought it might be a suitable name for this purpose.  Another one I was thinking of was `Strand` but that seemed even more esoteric.

I'm open to future naming changes if desired but I _think_ this is sufficient for now?